app/bundles/CoreBundle/Model/FormModel.php

```diff
@@ -50,10 +50,24 @@ public function isLocked($entity)
     {
         if (method_exists($entity, 'getCheckedOut')) {
             $checkedOut = $entity->getCheckedOut();
-            if (!empty($checkedOut)) {
-                //is it checked out by the current user?
+            if (!empty($checkedOut) && $checkedOut instanceof \DateTime) {
                 $checkedOutBy = $entity->getCheckedOutBy();
-                if (!empty($checkedOutBy) && $checkedOutBy !== $this->userHelper->getUser()->getId()) {
+                $maxLockTime  = $this->factory->getParameter('max_entity_lock_time', 0);
```

Comment:
@escopecz 

Well, I saw mautic factory was deprecated, but also I saw the comment at the top of factory class :
 `  Do not use Factory in Models. There's a couple places where we still need to in core, but we are working on refactoring. This is completely temporary.`
I tought I was in this case.

The core class `FormModel` is extended in 26 other class, if I add a contructor to this class, i need to modify and test 26 other class... it look like a very big modification.